docs/TuxArchitecture.md

```diff
@@ -0,0 +1,186 @@
+#Tux/Architecture
+
+##Table of Contents
+<ol>
+  <li><a href="#Premise">Premise</a></li>
+  <li><a href="#Implementation">Implementation</a>
+    <ol style="list-style-type:upper-alpha">
+      <li><a href="#Architecting-Store-Dependencies">Architecting Store Dependencies</a>
+        <ol style="list-style-type:upper-roman">
+          <li><a href="#Architecting-Store-Dependencies-Example">Architecting Store Dependencies Example</a></li>
+          <li><a href="#Architecture-architect">METHOD: Architecture.architect</a></li>
+          <li><a href="#architectureChain">OBJECT: architectureChain</a>
+            <ol style="list-style-type:lower-alpha">
+              <li><a href="#architectureChain-itNeeds">METHOD: architectureChain.itNeeds</a></li>
+              <li><a href="#architectureChain-itOutputs">METHOD: architectureChain.itOutputs</a></li>
+              <li><a href="#architectureChain-and">METHOD: architectureChain.and</a></li>
+            </ol>
+          </li>
+        </ol>
+      </li>
+    </ol>
+  </li>
+</ol>
+
+##<a id="Premise"></a>Premise [#](#Premise)
+>TuxedoJS provides a novel approach to describing the dependencies between stores. In Flux, store dependencies are defined via the `Dispatcher.waitFor` syntax. This allows stores to define the order at which they will receive events from the dispatcher. While this syntax is workable, we consider it to be flawed due to the fact that it requires each store to separately manage the stores it needs to waitFor. This results in distributed app architectures that are difficult to reason about. For an engineer to understand all the store relationships in an app, he/she would need to step through every store file and draw out the map of stores it is dependent on. TuxedoJS tackles this problem by centralizing waitFor statements and providing a more semantic interface for building waitFor relationships.
```

Comment:
Recommend changing "This results in distributed app architectures that are difficult to reason about." to "This can result in distributed app architectures that are difficult to reason about."